### PR TITLE
Default to using touchend for Handler.Feature, not touchstart

### PR DESCRIPTION
**Pull request by davea**
This is an existing PR rebased on our `master` branch: https://github.com/openlayers/ol2/pull/1118 - no other changes made. [MS I made some other changes]

Compiled OL ready for FMS testing in this PR: https://github.com/mysociety/fixmystreet/pull/4395

### CODE REVIEW

**Comment by davea** [comment]
Unfortunately this now means that if you attempt to pan the map and your touch starts on a marker nothing will happen.

You have to tap on an empty bit of the map, which is problematic in really busy areas, e.g.:

Screenshot 2023-04-20 at 16 00 18

I briefly investigated how to allow panning in that situation but didn't get very far. I wonder if we could do something to make the tap targets smaller on mobile, e.g. removing the shadows or shrinking the markers a little bit...

**Review by dracos — APPROVED** [review]
Not sure what the first commit is actually doing, the change doesn't require the user of the featureOptions/boxOptions, looks like it might be for the tests I guess? Anyway, if it fixes the issue, woo!

**Comment by dracos** [comment]
"Unfortunately this now means that if you attempt to pan the map and your touch starts on a marker nothing will happen." - sigh

**Comment by dracos** [comment]
So we want to allow the pan to carry on wherever, what we don't want is the selection of the object to happen if it's a pan.

**Comment by dracos** [comment]
Weird, how I read the code is that the dragging should be unaffected, only the selection. Wonder what's going on.

**Comment by davea** [comment]
I mean it’s still better than the current behaviour of a random report appearing if you start on a marker 🙂

**Comment by dracos** [comment]
Yes, agreed it is better! I'll take a look and see if I can spot anything obvious.

File: lib/OpenLayers/Handler/Feature.js [hunk 1]
```diff
@@ -132,11 +140,51 @@ OpenLayers.Handler.Feature = OpenLayers.Class(OpenLayers.Handler, {
      * {Boolean} Let the event propagate.
      */
     touchstart: function(evt) {
-        this.startTouch(); 
+    	// do we callback on the touchstart or on the touchend event?
+    	if (this.callbackOnTouchend) {
+    		this.EVENTMAP.touchstart = this.EVENTMAP.mousedown;
```
Comment (dracos) [inline lib/OpenLayers/Handler/Feature.js]: EVENTMAP is global, I think, so this is changing it for every Feature Handler, presumably depending upon which one is last set up.

Comment (dracos) [reply lib/OpenLayers/Handler/Feature.js]: have pushed a fix for this little thing I think

File: lib/OpenLayers/Handler/Feature.js [hunk 2]
```diff
@@ -94,7 +102,7 @@ OpenLayers.Handler.Feature = OpenLayers.Class(OpenLayers.Handler, {
      *      mousedowns do propagate. Unhandled mousedowns always propagate,
      *      whatever the value of stopDown. Defaults to true.
      */
-    stopDown: true,
+    stopDown: false,
```
Comment (davea) [inline lib/OpenLayers/Handler/Feature.js:105]: For the sake of completeness I guess the comment on L103 should reflect the new default

**Pull request merged**